junit-jupiter-api/src/main/java/org/junit/jupiter/api/ClassOrderer.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api;
+
+import static java.util.Comparator.comparingInt;
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Optional;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+
+/**
+ * {@code ClassOrderer} defines the API for ordering the <em>top-level test classes</em>,
+ * without considering nested test classes.
+ *
+ * <p>In this context, the term "test class" refers to any class containing methods annotated with
+ * {@code @Test}, {@code @RepeatedTest}, {@code @ParameterizedTest},
+ * {@code @TestFactory}, or {@code @TestTemplate} and not being annotated with {@code @Nested}.
+ *
+ * <h4>Built-in Implementations</h4>
+ *
+ * <p>JUnit Jupiter provides the following built-in {@code ClassOrderer}
+ * implementations.
+ *
+ * <ul>
+ * <li>{@link ClassOrderer.ClassName}</li>
+ * <li>{@link ClassOrderer.DisplayName}</li>
+ * <li>{@link ClassOrderer.OrderAnnotation}</li>
+ * <li>{@link ClassOrderer.Random}</li>
+ * </ul>
+ *
+ * @see ClassOrdererContext
+ * @see #orderClasses(ClassOrdererContext)
+ * @since 5.8
+ */
+@API(status = EXPERIMENTAL, since = "5.8")
+public interface ClassOrderer {
+
+	/**
+	 * Property name used to set a custom {@link ClassOrderer} implementation: {@value}
+	 *
+	 * <h3>Supported Values</h3>
+	 *
+	 * <p>Supported values include fully qualified class names for types that
+	 * implement {@link org.junit.jupiter.api.ClassOrderer}.
+	 */
+	String CUSTOM_CLASS_ORDERER_PROPERTY_NAME = "junit.jupiter.execution.class.order.config.custom.class";
```

Comment:
Let's rename this to `junit.jupiter.testclass.order.default` for consistency with test method ordering and inline it in the corresponding constant in `JupiterConfiguration`. The `default` is a bit superfluous at the moment but we may decide to add support for ordering nested classes or a package-level annotation for top-level classes.